packages/python/plotly/_plotly_utils/basevalidators.py

```diff
@@ -103,6 +103,13 @@ def copy_to_readonly_numpy_array(v, kind=None, force_numeric=False):
             else:
                 # DatetimeIndex
                 v = v.to_pydatetime()
+    elif pd and isinstance(v, pd.DataFrame) and len(set(v.dtypes)) == 1:
```

Comment:
this wouldn't handle a `customdata` case where all the custom data isn't the same type, right?

Reply:
Right, but that was already working (in my testing)